src/operations/operation.ts

```diff
@@ -64,20 +83,8 @@ export abstract class OperationBase<
     return ctor.aspects.has(aspect);
   }
 
-  set session(session: ClientSession) {
-    Object.assign(this.options, { session });
-  }
-
-  get session(): ClientSession {
-    // NOTE: Using the bang operator here because we know there is always a
-    //       session, explicit or implicit. We should disambiguate the session
-    //       from the options and set it as an explicit field
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    return this.options.session!;
-  }
-
   clearSession(): void {
-    delete this.options.session;
+    delete this.session;
```

Comment:
Echoing @reggi's comments, do want to use `delete` here or should we set it to `undefined`?